### PR TITLE
[Cloudflare] Add _id field to LogPull data_stream

### PR DESCRIPTION
**Pull request by legoguy1000**
## What does this PR do?

adds teh fingerprint processor to deduplicate events for Cloudflare LogPull when using multiple agents.

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

**Comment by efd6** [comment]
Please run `elastic-package build`.

**Comment by legoguy1000** [comment]
> Please run `elastic-package build`.

I did multiple times and it hasn't made any changes.  I normally use the latest Elastic Package build from main branch and even changed to the latest tag, v0.47.0 and still no changes from whats already in the branch.

**Comment by andrewkroh** [comment]
> I did multiple times and it hasn't made any changes. I normally use the latest Elastic Package build from main branch and even changed to the latest tag, v0.47.0 and still no changes from whats already in the branch.

Double check the version you have installed with `elastic-package version`. From this branch it had changes for me. I used:

```
gh pr checkout 3187
go install github.com/elastic/elastic-package
cd packages/cloudflare
elastic-package build
```

File: packages/cloudflare/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.4.1"
+  changes:
+    - description: Add "_id" field to LogPull data_stream deduplicate events
```
Comment (andrewkroh) [inline packages/cloudflare/changelog.yml:4]:
```suggestion
    - description: Add `_id` field to the logpull data stream to deduplicate events.
```

**Comment by legoguy1000** [comment]
@andrewkroh Ya installing elastic package from the `go install` command instead of building from source appeared to do it.

**Review by andrewkroh — APPROVED** [review]

**Comment by CyberTaoFlow** [comment]
How exactly does fingerprint de-duplicate events when using a data-stream?    My experience has historically (before data-streams) been that the fingerprint becomes the document id which you then either upsert or update any existing document with that ID.   

However with data-streams that is not going to work as the actions are limited to create when targeting the data-stream alias right?

From the elasticsearch datastream documentation...
```
If needed, you can update or delete documents by submitting requests directly to the document’s backing index.

**Pull request merged**